### PR TITLE
V2 : Correcting use of 'size' parameter in primitives 

### PR DESCRIPTION
**Pull request by z3dev**
This pull request corrected the use / meaning of the 'size' parameter in the following primitives. Size now means full dimension, i.e. full width, length, depth, and height.
- cube
- cuboid
- rectangle
- roundedCuboid
- roundedRectangle
- square

All test suites that use these primitives have been adjusted as well.

### CODE REVIEW

**Comment by z3dev** [comment]
As discussed, I have made changes to correct the use / meaning of 'size'.

@kaosat-dev please review and merge

File: src/primitives/rectangle.js [hunk 1]
```diff
@@ -2,19 +2,20 @@ const vec2 = require('../math/vec2')
 
 const geom2 = require('../geometry/geom2')
 
-/** Construct an axis-aligned rectangle with four sides and four 90-degree angles.
+/**
+ * Construct an axis-aligned rectangle with four sides and four 90-degree angles.
  * @param {Object} [options] - options for construction
  * @param {Array} [options.center=[0,0]] - center of rectangle
- * @param {Array} [options.size=[1,1]] - size of rectangle, width and height
+ * @param {Array} [options.size=[2,2]] - full dimension of rectangle, width and length
```
Comment (kaosat-dev) [inline src/primitives/rectangle.js]: I do not think it is necessary to specify 'full dimensions' here, that is the logical assumption when you say width and length I think :)

File: src/primitives/cuboid.js [hunk 2]
```diff
@@ -12,15 +13,15 @@ const {geom3, poly3} = require('../geometry')
 const cuboid = (options) => {
   const defaults = {
     center: [0, 0, 0],
```
Comment (kaosat-dev) [inline src/primitives/cuboid.js:15]: this is odd, I had not noticed the addition of center to all these ?

**Review by kaosat-dev — CHANGES_REQUESTED** [review]
Looking good ! Just very minor tweak and merge is go !

File: src/primitives/cuboid.js (hunk 2 above)
Comment (z3dev) [reply src/primitives/cuboid.js:15]: Yup. Center is part of all primitives except cylinders.

It wasn't added. 'center' has always been part of the functionality, and came with the conversion of V1 to V2.

File: src/primitives/rectangle.js (hunk 1 above)
Comment (z3dev) [reply src/primitives/rectangle.js]: Done

**Comment by z3dev** [comment]
This should be ready to go now.

@kaosat-dev please merge.

**Review by kaosat-dev — APPROVED** [review]

**Pull request merged**